### ISSUE TITLE
Splitting+replacing token does not work as expected

### ISSUE THREAD

**Issue by pgolo** [pgolo/sic#32]
Scenario: we want to transform string so that plurals are turned into singulars:
`plurals` -> `plural`
`plurals plurals` -> `plural plural`

For this purpose, the tokenizer config is supposed to be as follows:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<tokenizer name="test_plural">
  <setting name="cs" value="0" />
  <split where="r" value="s" />
  <token to="" from="s" />
</tokenizer>
```

As of version 1.0.4, such configuration produces faulty output:
`plurals` -> `plural s`
`plurals plurals` -> `plural s plura s`

**Comment by pgolo** [comment]
Fixed with MR #36

**Comment by pgolo** [comment]
Now it replaces substring even if it is not token (no instruction to split it is given):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<tokenizer name="test_plural">
  <setting name="cs" value="0" />
  <token to="" from="s" />
</tokenizer>
```
As of version 1.0.5, such configuration produces faulty output:
`plurals` -> `plural`
`plurals plurals` -> `plural plural`

**Comment by pgolo** [comment]
Fixed with MR #38
